### DISCUSSION TITLE
How to configure everything?

### DISCUSSION THREAD

**Discussion by maehw** [xtech/self-o-mat#9] [Q&A] [answered]
I've started to document the configuration possibilities here: https://github.com/xtech/self-o-mat/wiki/Configuration

1. How do templates work? Are templates used to customize the UI or the photo printouts? Are there example files that can be used - also for documentation?
2. What does "filtering" refer to?
3. i18n: I've only seen `*_frontend.json` being used so far, what are `en.json` and `de.json` being used for?
4. Instead of printing every photo by default when the printer is enabled (opt-out) - is there a way to switch it around so that photos can be printed on request until a new one is taken (opt-in)?
5. Is there a way to specify a printer (by name or connection)? As at least during development one might have attached one's normal printer and a photo printer.
6. Is the app mentioned on the website included in this repo? Or is it just a website offered by a webserver running with **self-o-mat**? Are these settings accessible from a different interface as well?

Edit: oops, the German website https://www.self-o-mat.de/ (where you could buy kits) gives some more ideas about filters and templates.

Cheers

**Comment by ClemensElflein** [comment, marked as answer]
> 
> 
> I've started to document the configuration possibilities here: https://github.com/xtech/self-o-mat/wiki/Configuration
> 
>     1. How do templates work? Are templates used to customize the UI or the photo printouts? Are there example files that can be used - also for documentation?

Templates are basically overlay over the printed version of the photo. The idea is that you can frame the picture or overlay the final picture with a company logo.
In order to create a template, you simply need to create a PNG frame with a transparent region. The photo booth software will detect the transparent region and insert the image there. Note that the image will be scaled in such a way that it's not stretched but covers the whole area.

An example is here:
![Photo booth Template Example](https://help.self-o-mat.de/uploads/images/gallery/2020-03/Dyk2oc7EwpYWCHr7-template.png)

You can download a template here:
[Template](https://help.self-o-mat.de/downloads/template_leer.png)

> 
>     2. What does "filtering" refer to?

Some users noted that the images looked "boring" and not saturated enough. This is the case because smartphones tend to over-saturate images and add contrast whereas DSLRs don't do that. If the filter is enabled, the final picture gets passed through an image processing filter which adds contrast and saturation for "better" photos. (I'm not really a fan of the option but many like it 🤷‍♂️ )

> 
>     3. i18n: I've only seen `*_frontend.json` being used so far, what are `en.json` and `de.json` being used for?

The *_frontend.json files are for the photo booth's screen, whereas the en.json and de.json are for the browser accessible settings app.

> 
>     4. Instead of printing every photo by default when the printer is enabled (opt-out) - is there a way to switch it around so that photos can be printed on request until a new one is taken (opt-in)?

This is currently not implemented, sorry.

> 
>     5. Is there a way to specify a printer (by name or connection)? As at least during development one might have attached one's normal printer and a photo printer.

The app prints to CUPS' default destination. Just set the default printer to the one you like.

> 
>     6. Is the app mentioned on the website included in this repo? Or is it just a website offered by a webserver running with **self-o-mat**? Are these settings accessible from a different interface as well?

We had a native app, but it was too hard to maintain. Therefore the "app" mentioned on the website refers to the webserver running on the self-o-mat.

> 
> 
> Edit: oops, the German website https://www.self-o-mat.de/ (where you could buy kits) gives some more ideas about filters and templates.

No problem, do you by any chance understand German? We have started to document the self-o-mat on [https://help.self-o-mat.de/](https://help.self-o-mat.de/).
